grammar.js

```diff
@@ -630,21 +634,45 @@ module.exports = grammar(C, {
 
     lambda_expression: $ => seq(
       field('captures', $.lambda_capture_specifier),
-      field('declarator', $.abstract_function_declarator),
+      optional(field('declarator', $.abstract_function_declarator)),
       field('body', $.compound_statement)
     ),
 
     lambda_capture_specifier: $ => prec(PREC.LAMBDA, seq(
       '[',
       choice(
         $.lambda_default_capture,
-        commaSep($._expression)
+        commaSep($._expression),
+        seq(
+          $.lambda_default_capture,
+          ',', commaSep1($._expression)
+        )
       ),
       ']'
     )),
 
     lambda_default_capture: $ => choice('=', '&'),
 
+    parameter_pack_expansion: $ => prec(-1, seq(
+      field('pattern', $._expression),
+      '...'
+    )),
+
+    type_parameter_pack_expansion: $ => seq(
+      field('pattern', $.type_descriptor),
+      '...'
+    ),
+
+    sizeof_expression: ($, original) => choice(
+      original,
+      seq(
+        'sizeof', '...',
```

Comment:
I opted to separate `sizeof` and `...` tokens so they would match other rules, instead of making a brand new `sizeof...` token. If you think it'd be more appropriate to:
1. use a `sizeof...` token directly; or
2. make an entirely new rule (`variadic_sizeof_expression`)

then just let me know and I can make it happen.